modules/hdf5_handler/h5commoncfdap.cc

```diff
@@ -1632,6 +1634,113 @@ void add_var_dap4_attr(BaseType *var,const string& attr_name, D4AttributeType at
 
 }
 
+// Add DAP4 coverage 
+void add_dap4_coverage(libdap::D4Group* d4_root, const vector<string>& coord_var_names, bool is_coard) {
+
+    // We need to construct the var name to Array map,using unordered_map for quick search.
+    unordered_map<string, Array*> d4map_array_maps;
+
+    // This vector holds all variables that can have coverage maps.
+    vector<Array*> has_map_arrays;
+  
+    Constructor::Vars_iter vi = d4_root->var_begin();
+    Constructor::Vars_iter ve = d4_root->var_end();
+
+    for (; vi != ve; vi++) {
+
+        BaseType *v = *vi;
+
+        // Only Array can have maps.
+        if (libdap::dods_array_c == v->type()) {
+
+            Array *t_a = static_cast<Array *>(*vi);
+
+            // The maps are essentially coordinate variables.
+            // We've sorted the coordinate variables already, so
+            // just save them to the vector. 
+            // Note: the coordinate variables we collect here are
+            // based on the understanding of the handler. We will
+            // watch if there are complicated cases down the road. KY 04-15-2022
+            bool is_cv = false;
+            for ( auto it_cv = coord_var_names.begin(); it_cv !=coord_var_names.end();++it_cv) {
```

Comment:
For loops that use iterators with '.begin()' and '.end()' methods, you can use the shorter:
for (auto cv : coard_cv_names) {
     ...
}

(another C++-11 feature).

Then, line 1666 becomes
if (cv == v->name()) { 
...
}

Reply:
Thanks, will handle this later.